### PR TITLE
DSND-627: Removed transactional configuration

### PR DESCRIPTION
**Pull request by vchitti-kainos**

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java [hunk 1]
```diff
@@ -86,9 +86,14 @@ public void updateInsolvencyLink(String contextId, String companyNumber,
...
         try {
 
-            CompanyProfileDocument cpDocument =
-                    companyProfileRepository.findById(companyNumber).get();
+            Optional<CompanyProfileDocument> cpDocumentOptional =
+                    companyProfileRepository.findById(companyNumber);
+
+            cpDocumentOptional.orElseThrow(() -> new IllegalArgumentException(
+                    String.format("No company profile with company number %s found",
+                    companyNumber)));
```
Comment (iichr) [inline src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java:94]: I believe this needs to be thrown as a `BadRequestException` as it was originally (mainly due to the status code expected)

File: src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java [hunk 2]
```diff
@@ -102,17 +107,22 @@ public void updateInsolvencyLink(String contextId, String companyNumber,
...
+            }
         } catch (DataAccessException dbException) {
             throw new ServiceUnavailableException(dbException.getMessage());
         }
         logger.trace(String.format("Company profile is updated in MongoDB with contextId %s "
                 + "and company number %s", contextId, companyNumber));
-
-        insolvencyApiService.invokeChsKafkaApi(contextId, companyNumber);
-        logger.info(String.format("Chs-kafka-api CHANGED invoked successfully for "
-                + "contextId %s and company number %s", contextId, companyNumber));
```
Comment (iichr) [inline src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java:116]: This log is specifically added as part of a logging requirement - please keep and move immediately after the successful `invokeChsKafkaApi` call.

File: src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java [hunk 3]
```diff
@@ -102,17 +107,22 @@ public void updateInsolvencyLink(String contextId, String companyNumber,
...
+                    contextId, companyNumber);
+
+            if (response.getStatusCode() == HttpStatus.OK.value()) {
+                companyProfileRepository.save(cpDocument);
+            } else {
+                String errorMessage = String.format(
+                        "Error occurred while calling /resource-changed with "
+                        + "contextId %s and company number %s", contextId, companyNumber);
+                logger.error(errorMessage);
+                throw new RuntimeException(errorMessage);
```
Comment (iichr) [inline src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java:120]: All of this and more already happens in `src/main/java/uk/gov/companieshouse/company/profile/api/InsolvencyApiService.java` in the `invokeChsKafkaApi` method, no need to log and throw the same exception again.

**Review by iichr — CHANGES_REQUESTED** [review]
Just a few bits that need addressed to preserve existing behaviour

**Review by amobarak — APPROVED** [review]

**Review by iichr — APPROVED** [review]

**Pull request merged**